### PR TITLE
TM-3706 Edit Position Details

### PR DESCRIPTION
**Pull request by andrews-john**
Initial commit. Added changes for created-modified by.

### CODE REVIEW

File: src/Components/BureauPage/PositionManager/PositionManager.jsx [hunk 1]
```diff
@@ -319,7 +319,7 @@ const PositionManager = props => {
     selectedOrgs,
     selectedBureaus,
   ]);
-
+  console.log('bureauresults', bureauPositions.results);
```
Comment (SidelineCory24) [inline src/Components/BureauPage/PositionManager/PositionManager.jsx:322]: remove🧼

File: src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  result: POSITION_DETAILS.isRequired,
+  fromPostMenu: PropTypes.bool,
+};
+
+PositionDetailCard.defaultProps = {
+  isProjectedVacancy: false,
+  fromPostMenu: false,
+};
+
+export default PositionDetailCard;
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx:129]: update to to Details instead of Detail

File: src/Components/PositionDetails/PositionDetails.jsx [hunk 3]
```diff
@@ -86,6 +86,7 @@ class PositionDetails extends Component {
             userProfile={userProfile}
             isProjectedVacancy={isProjectedVacancy}
           />
+          wefoinwiowenf
```
Comment (SidelineCory24) [inline src/Components/PositionDetails/PositionDetails.jsx:89]: 🧼

File: src/actions/panelMeetingAgendas.js [hunk 4]
```diff
@@ -31,7 +31,7 @@ const dummyAgenda = [
         neuid: 87496,
         last_name: 'Andrews',
         first_name: 'John',
-        neumiddlenm: 'Cléopatre',
+        middle_name: 'Philip',
```
[the remark this answers is not in the document]
Comment (andrews-john) [reply src/actions/panelMeetingAgendas.js:34]: fair enough lmao

File: src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx [hunk 5]
```diff
@@ -95,8 +95,8 @@ const AgendaItemRow = props => {
               </div>
             }
             <div>
-              <div className="label">Created By: <span>{get(agenda, 'creators.first_name' || 'Default')} {get(agenda, 'creators.last_name' || 'Default')}</span></div>
-              <div className="label">Modified By: <span>{get(agenda, 'updaters.first_name' || 'Default')} {get(agenda, 'updaters.last_name' || 'Default')}</span></div>
+              <div className="label">Created By: <span>{get(agenda, 'creators.last_name' || 'Default')}, {get(agenda, 'creators.first_name' || 'Default')} {get(agenda, 'creators.middle_name' || 'Default').charAt(0)}</span></div>
+              <div className="label">Modified By: <span>{get(agenda, 'updaters.last_name' || 'Default')}, {get(agenda, 'updaters.first_name' || 'Default')} {get(agenda, 'updaters.middle_name' || 'Default').charAt(0)}</span></div>
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx:99]: can revert these changes

Comment (andrews-john) [reply src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx:99]: Which changes? The get() is required to pass the yarn tests. The change to (lastname, firstname middlename) was a direct ask from Kate, as thats how it appears in the ticket

File: src/Components/EditPositionDetails/EditPositionDetails.jsx [hunk 6]
```diff
@@ -213,6 +363,14 @@ const EditPositionDetails = () => {
             </div>
           }
         </div>
+        <div className="usa-width-one-whole position-manager-lower-section results-dropdown">
+          <div className="usa-grid-full position-list">
+            {<PositionDetailsCard
+              result={dummyPositionDetails}
+              key={dummyPositionDetails.id}
+            />}
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/EditPositionDetails.jsx:371]: can remove curly braces

File: src/Components/PositionDetails/PositionDetails.jsx [hunk 7]
```diff
@@ -86,6 +86,7 @@ class PositionDetails extends Component {
             userProfile={userProfile}
             isProjectedVacancy={isProjectedVacancy}
           />
+
```
Comment (SidelineCory24) [inline src/Components/PositionDetails/PositionDetails.jsx:89]: 🧼

File: src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx (hunk 5 above)
Comment (SidelineCory24) [reply src/Components/Agenda/AgendaItemRow/AgendaItemRow.jsx]: discussed offline, updates made

File: src/actions/panelMeetingAgendas.js [hunk 8]
```diff
@@ -21,17 +21,17 @@ const dummyAgenda = [
...
         hruempseqnbr: null,
         hruneuid: 87496,
         hruid: 65426,
         neuid: 87496,
-        last_name: 'Andrews',
-        first_name: 'John',
-        neumiddlenm: 'Cléopatre',
+        last_name: 'Bob',
+        first_name: 'Rob',
+        middle_name: 'P',
```
Comment (SidelineCory24) [inline src/actions/panelMeetingAgendas.js:34]: 😂

File: src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.test.jsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { shallow } from 'enzyme';
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.test.jsx:1]: nice work for your first test!

File: src/Components/EditPositionDetails/EditPositionDetails.jsx [hunk 10]
```diff
@@ -213,6 +363,14 @@ const EditPositionDetails = () => {
             </div>
           }
         </div>
+        <div className="usa-width-one-whole position-manager-lower-section results-dropdown">
+          <div className="usa-grid-full position-list">
+            <PositionDetailsCard
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/EditPositionDetails.jsx:368]: 😎

File: src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+const PositionDetailsCard = ({ result, isProjectedVacancy, fromPostMenu }) => {
+  const [showMore, setShowMore] = useState(false);
+
+
+  const pos = get(result, 'position') || result;
+
+  const title = propOrDefault(pos, 'title');
+  const position = getResult(pos, 'position_number', NO_POSITION_NUMBER);
+  const languages = getResult(pos, 'languages', []);
+  const hasShortList = getResult(result, 'has_short_list', false);
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx:30]: can remove `hasShortList` and where it's being used

File: src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+import { getDifferentials, getResult } from 'Components/ResultsCard/ResultsCard';
+import LanguageList from 'Components/LanguageList';
+import { getPostName, propOrDefault, shortenString } from 'utilities';
+import {
+  NO_BUREAU, NO_DATE, NO_GRADE,
+  NO_POSITION_NUMBER, NO_POST, NO_SKILL, NO_TOUR_OF_DUTY, NO_UPDATE_DATE, NO_USER_LISTED,
+} from 'Constants/SystemMessages';
+import { POSITION_DETAILS } from 'Constants/PropTypes';
+
+const PositionDetailsCard = ({ result, isProjectedVacancy, fromPostMenu }) => {
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx:21]: can remove fromPostMenu

Comment (SidelineCory24) [reply src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx:21]: can remove isProjectedVacancy

File: src/Components/EditPositionDetails/EditPositionDetails.jsx [hunk 13]
```diff
@@ -136,6 +137,155 @@ const EditPositionDetails = () => {
...
 
+  const dummyPositionDetails = {
+    id: '2561',
+    status: null,
+    status_code: 'OP',
+    ted: '2025-05-12T21:12:12.854000Z',
+    posted_date: '2022-10-26T21:12:12.854000Z',
+    availability: {
+      availability: null,
+      reason: null,
```
Comment (pveland621) [inline src/Components/EditPositionDetails/EditPositionDetails.jsx:148]: @SidelineCory24 just wondering - how do we decide if dummy data should go into an action file or just live right here in the component file?

Comment (SidelineCory24) [reply src/Components/EditPositionDetails/EditPositionDetails.jsx:148]: nice catch, will address offline to put this in an action file

File: src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+import { getDifferentials, getResult } from 'Components/ResultsCard/ResultsCard';
+import LanguageList from 'Components/LanguageList';
+import { getPostName, propOrDefault, shortenString } from 'utilities';
+import {
+  NO_BUREAU, NO_DATE, NO_GRADE,
+  NO_POSITION_NUMBER, NO_POST, NO_SKILL, NO_TOUR_OF_DUTY, NO_UPDATE_DATE, NO_USER_LISTED,
+} from 'Constants/SystemMessages';
+import { POSITION_DETAILS } from 'Constants/PropTypes';
+
+const PositionDetailsCard = ({ result, isProjectedVacancy }) => {
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx:20]: can remove isProjectdVacany

File: src/actions/editPositionDetails.js [hunk 15]
```diff
@@ -1,10 +1,156 @@
...
+  bidcycle: {
+    id: 160,
+    name: 'Details/Training 2019',
+    cycle_start_date: null,
+    cycle_deadline_date: null,
+    cycle_end_date: null,
+    active: null,
+    handshake_allowed_date: null,
+  },
+  bid_statistics: [
```
Comment (SidelineCory24) [inline src/actions/editPositionDetails.js:121]: can remove `bid_statistics`

File: src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+  const position = getResult(pos, 'position_number', NO_POSITION_NUMBER);
+  const languages = getResult(pos, 'languages', []);
+
+  const language = (<LanguageList languages={languages} propToUse="representation" />);
+
+  const postShort = `${getPostName(get(pos, 'post') || NO_POST)}`;
+  const description = shortenString(get(pos, 'description.content') || 'No description.', 2000);
+  const id = get(result, 'id') || '';
+
+  const detailsLink = (<Link to={`/profile/bureau/positionmanager/available/${id}`}>
```
Comment (SidelineCory24) [inline src/Components/EditPositionDetails/PositionDetailsCard/PositionDetailsCard.jsx:35]: I jumped the line on this, you can add this back
```
const detailsLink = (<Link to={`/profile/bureau/positionmanager/${isProjectedVacancy ? 'vacancy' : 'available'}/${result.id}`}>
    <h3>{title}</h3></Link>);
```

as well as the `isProjectedVacancy` Prop

**Review by SidelineCory24 — APPROVED** [review]
Looks good to go!🚀
